src/templates/tag.js

```diff
@@ -49,52 +45,87 @@ function getTagDependencies(tagStruct, tags) {
   }));
 }
 
-module.exports = (page, metaIndex) => {
-  const tagClassSnake = page._pathParts[page._pathParts.length - 1];
-  const tagClassPascal = page.tagClass || tagClassSnake
+function getChildTags(tagStruct, tags) {
+  return Object.values(tags).filter(otherTagStruct =>
+    //Invader's BasicObject is the only inherited struct which is not a traditional tag type
+    otherTagStruct.name != "BasicObject" && (
+      otherTagStruct.inherits == tagStruct.name ||
+      tagStruct.name == "Object" && otherTagStruct.inherits == "BasicObject"
+    )
+  );
+}
+
+function getTagClassPascalCase(page) {
+  return page.tagClass || page._slug
     .split("_")
     .map(part => `${part[0].toUpperCase()}${part.substring(1)}`)
     .join("");
+}
```

Comment:
This is needed to convert the page slug (e.g. `sound_scenery`)  to a Pascal case struct name which matches Invader's defs (e.g. `SoundScenery`), so we can lookup fields and dependencies. This doesn't work for every slug, like `scenario_structure_bsp` needing BSP to be all upper-case, so the page can override with `tagClass` via its YAML prefix. This isn't new code, but just wanted to explain what it's doing and add context to my earlier comment about trying to simplify this later.